dnf-behave-tests/dnf/steps/fixtures/httpd.py

```diff
@@ -49,9 +49,12 @@ def do_GET(self):
         # process the request as usual.
         if 'status' in self.server._conf:
             self.send_response(self.server._conf['status'])
+            if "response_headers" in self.server._conf:
```

Comment:
I'd recommend to add an empty `"response_headers"` dict to `_conf` when initializing it, i.e. around line 102 in `HttpServerContext.__init__()` (if I'm not mistaken).

This check should instead be:
```python
if self.path in self.server._conf["response_headers"]:
```

Because you can have some response headers configured but they don't need to be for this path and then you'd get an exception.

Also, please don't add your block into the `if 'status' in self.server._conf:` block. The response headers should work even if there's no special status set. (if it doesn't work because setting the headers doesn't cooperate with the `super(HttpHandler, self).do_GET()` being called at the end, let me know and we can have a look at that together)